### ISSUE TITLE
PSSetShaderResources  Crashing

### ISSUE THREAD

**Issue by dubinder** [ocornut/imgui#1265]
I must not be doing something correctly.

I am crashing in imgui_impl_dx11.cpp line 207
```c++
ctx->PSSetShaderResources(0, 1, (ID3D11ShaderResourceView**)&pcmd->TextureId);
```
I am passing a shaderResesourceView to 
```c++
ImGui::Image((ImTextureID)(m_textureView), ImVec2(1920, 1080))
```

ShaderResouce Creation
```c++
D3D11_TEXTURE2D_DESC textureDesc;
  D3D11_SUBRESOURCE_DATA subData;
  D3D11_SHADER_RESOURCE_VIEW_DESC srvDesc;
  memset(&textureDesc, 0, sizeof(textureDesc));
  memset(&subData, 0, sizeof(subData));
  memset(&srvDesc, 0, sizeof(srvDesc));
  textureDesc.Width = 1920;
  textureDesc.Height = 1080;
  textureDesc.MipLevels = 1;
  textureDesc.ArraySize = 1;
  textureDesc.Format = DXGI_FORMAT_R8G8B8A8_UNORM;
  textureDesc.SampleDesc.Count = 1;
  textureDesc.Usage = D3D11_USAGE_DYNAMIC;
  textureDesc.BindFlags = D3D11_BIND_SHADER_RESOURCE;// | D3D11_BIND_RENDER_TARGET;
  textureDesc.CPUAccessFlags = D3D11_CPU_ACCESS_WRITE;
  textureDesc.MiscFlags = 0;

  subData.pSysMem = buffer;
  subData.SysMemPitch = 1920 * 4;

  result = d3d11_device->CreateTexture2D(&textureDesc, &subData, &m_texture);
  if (FAILED(result))
  {
    MessageBox(hwnd, "Could not Create Texture", "Error", MB_OK);
   return;
  }

  srvDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2D,
    srvDesc.Texture2D.MostDetailedMip = 0;
  srvDesc.Texture2D.MipLevels = textureDesc.MipLevels;
  srvDesc.Format = textureDesc.Format;
  result = d3d11_device->CreateShaderResourceView(m_texture, &srvDesc, &m_textureView);
  if (FAILED(result))
  {
    MessageBox(hwnd, "Could not create SRV", "Error", MB_OK);
    return;
  }
  
  m_texture->Release();
```

I keep getting Access violation reading location at some real small address.  I figure this must be from my lack of knowledge of dx11.  Is there something glaringly obvious that I am not doing correctly?

**Comment by ocornut** [comment]
Is your `m_textureView` pointer not-null ?
What happens if you call directly `ctx->PSSetShaderResources(0, 1, &m_textureView`)` in your code (as a dummy test, it won't have an effect on actual rendering obviously.

When you have a DirectX issue you can enable the debug layer and it generally will dump useful information in the Output window.

**Comment by dubinder** [comment]
m_textureView is not-null. 

I tried calling PSSetShaderResources, right after I created the  Shader Resource, and it doesn't crash.  

Debug output 
D3D11 CORRUPTION: ID3D11DeviceContext::PSSetShaderResources: Third parameter, array index 0 does not match device. [ MISCELLANEOUS CORRUPTION #15: CORRUPTED_PARAMETER3]

If this is because I am using my own device, and not the that is created in main.cpp, (g_pd3Device), That seems odd. I can give it a whirl see if that helps/fixes it.  This worked with dx9, just upgrading some of my projects to dx11.

**Comment by ocornut** [comment]
I assume you have only one single device and not actually running the code from main.cpp, since your application is already creating a device? You can pass your device to `ImGui_ImplDX11_Init()`.

**Comment by dubinder** [comment]
I will have to go back and look at some of the old d3d9 code, but I believe we were creating our own device, using that to create textures, and then passing those textures to ImGui::Image, and using your device and the textures worked fine.  I will have to check on that tomorrow.   

This may have been the issue(two d3d11 devices), because I got a quick test program up and running using your device to create the texture and ShaderResourceView, and I got my sweet Magenta test texture.  

And yes I totally overlooked passing our own device to ImGui_ImplDx11_Init.

If two devices worked in d3d9, and not in d3d11 I find that really interesting.  I will have to take a closer look at, that d3d9 code to see what we where doing. 

Thanks for such a great and useful library!
